File: app/src/main/java/cat/xojan/random1/domain/entities/EventLogger.java
package cat.xojan.random1.domain.entities;

import com.crashlytics.android.answers.Answers;
import com.crashlytics.android.answers.ContentViewEvent;
import com.crashlytics.android.answers.CustomEvent;

public class EventLogger {

    private Answers mAnswers;

    public EventLogger() {
    }

    public EventLogger(Answers answers) {
        mAnswers = answers;
    }

    public void logDownloadedPodcast(String title) {
        if (mAnswers != null) {
            mAnswers.logCustom(new CustomEvent("Downloads").putCustomAttribute("title", title));
        }

    }

    public void logPlayedPodcast(Podcast podcast) {
        if (mAnswers != null) {
            mAnswers.logContentView(new ContentViewEvent().putContentName(podcast.getTitle()));
        }
    }

    public void logExportedPodcast(final String podcastTitle) {
        if (mAnswers != null) {
            mAnswers.logCustom(new CustomEvent("Exported").putCustomAttribute("title", podcastTitle));
        }
    }

    public void logExportedPodcastAction() {
        if (mAnswers != null) {
            mAnswers.logCustom(new CustomEvent("ExportAction"));
        }
    }
}

File: app/src/main/java/cat/xojan/random1/domain/entities/EventLogger.kt
package cat.xojan.random1.domain.entities

import com.crashlytics.android.answers.Answers
import com.crashlytics.android.answers.ContentViewEvent
import com.crashlytics.android.answers.CustomEvent


class EventLogger(val answers: Answers?) {

    fun logDownloadedPodcast(title: String) {
        answers?.logCustom(CustomEvent("Downloads").putCustomAttribute("title", title))

    }

    fun logPlayedPodcast(podcast: Podcast) {
        answers?.logContentView(ContentViewEvent().putContentName(podcast.title))
    }

    fun logExportedPodcast(podcastTitle: String) {
        answers?.logCustom(CustomEvent("Exported").putCustomAttribute("title", podcastTitle))
    }

    fun logExportedPodcastAction() {
        answers?.logCustom(CustomEvent("ExportAction"))
    }
}